x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditUtil.java

```diff
@@ -38,4 +43,17 @@ public static String restRequestContent(RestRequest request) {
     private static Set<String> arrayToSetOrNull(String[] indices) {
         return indices == null ? null : new HashSet<>(Arrays.asList(indices));
     }
+
+    public static String getOrGenerateRequestId(ThreadContext threadContext) {
+        String requestId = extractRequestId(threadContext);
+        if (Strings.isEmpty(requestId)) {
```

Comment:
I think we can get rid of this.

Reply:
As in: be explicit when we generate these.
I know it would be a bummer if we hit an assertion about an event not assigned to a request, but I think there's value if we catch such an error, because such an event it might reveal hidden bugs.

Reply:
It's not really feasible to do that unfortunately.
When we process a transport action, we simply don't/can't know whether that action was spawned from an existing rest/transport request or whether it's standalone.

Well, I guess we could refactor a lot of stuff to make that explicit, but it's not right now.
_Maybe_, between system context, and _withOrigin_ we could pull something together to know when a transport action is isolated from any originating request, but it's going to be fragile.